crates/codec/src/ext.rs

```diff
@@ -1,42 +1,30 @@
-use std::io::{Cursor, Read, Result};
-use std::string::FromUtf8Error;
+use std::io::{Cursor, Read};
 
-use svm_types::{Address, State, TemplateAddr, TransactionId};
+use crate::EofError;
 
 /// A trait to be implemented by Decoders
-pub trait ReadExt {
-    /// Reads a single byte
-    fn read_byte(&mut self) -> Result<u8>;
-
-    /// Reads `length` bytes
-    fn read_bytes(&mut self, length: usize) -> Result<Vec<u8>>;
-
-    /// Reads a boolean
-    fn read_bool(&mut self) -> Result<bool>;
-
-    /// Reads an unsigned 16-bit integer (Big-Endian)
-    fn read_u16_be(&mut self) -> Result<u16>;
-
-    /// Reads an unsigned 32-bit integer (Big-Endian)
-    fn read_u32_be(&mut self) -> Result<u32>;
+pub trait ReadExt: Sized {
+    /// Tries to read the next byte, but doesn't move the cursor forward.
+    fn seek_byte(&self) -> Option<u8>;
```

Comment:
I think that `peek_byte` would make a better term here

Reply:
Renamed it.